src/kinetics-observer.cpp

```diff
@@ -578,32 +673,39 @@ int KineticsObserver::setIMU(const Vector3 & accelero,
 
   BOOST_ASSERT(imu.time < k_data_ && "The IMU has been already set, use another number");
 
+  imu.num = num;
   imu.acceleroGyro.head<3>() = accelero;
   imu.acceleroGyro.tail<3>() = gyrometer;
   imu.covMatrixAccelero = acceleroCov;
   imu.covMatrixGyro = gyroCov;
 
   if(imuSensors_[num].time == 0) /// this is the first value for the IMU
   {
-    imu.kinematics = localKine;
-    BOOST_ASSERT(imu.kinematics.position.isSet() && imu.kinematics.orientation.isSet()
+    imu.userImuKinematics = userImuKinematics;
+    imu.centroidImuKinematics = LocalKinematics(convertUserToCentroidFrame_(imu.userImuKinematics, k_data_));
+    BOOST_ASSERT(imu.centroidImuKinematics.position.isSet() && imu.centroidImuKinematics.orientation.isSet()
                  && "The kinematics of the IMU is incorrectly initialized");
-    if(!imu.kinematics.linVel.isSet())
+    if(!imu.centroidImuKinematics.linVel.isSet())
     {
-      imu.kinematics.linVel.set().setZero();
+      imu.centroidImuKinematics.linVel.set().setZero();
     }
```

Comment:
Maybe it's to add a `setToZeroIfUnset()` method for this?

Reply:
We do this only here, if we need it elsewhere we will consider this change, thank you!